static/js/TopicEditor.jsx

```diff
@@ -81,52 +90,68 @@ const TopicEditor = ({origData, onCreateSuccess, close, origWasCat}) => {
          const postCategoryData = {topics: sortedSubtopics};
          requestWithCallBack({url, data: postCategoryData, setSavingStatus, redirect: () => window.location.href = "/topics"});
     }
-    const saveTopic = function () {
-        toggle();
-        let url = "";
+
+    const prepData = () => {
         let postData = {...data, "description": {"en": data.enDescription, "he": data.heDescription}, "title": data.enTitle,
             "heTitle": data.heTitle};
         if (isCategory) {
             postData = {...postData, "catDescription": {"en": data.enCategoryDescription, "he": data.heCategoryDescription}};
         }
-        postData.category = data.catSlug;
-
-        if (isNew) {
-          url = "/api/topic/new";
+        if (data?.era && Sefaria.util.inArray(data.era, Sefaria._eras) === -1) {
+            delete postData.era;
```

Comment:
in order to clean data before sending it out to the server, best to check that data has changed and then add a key to the post object. That way if a user wants to remove data or set it to an empty string it is possible. This can also be done generically and not on a specific element such as "era"